constraint/pkg/externaldata/request.go

```diff
@@ -0,0 +1,30 @@
+package externaldata
+
+// RegoRequest is the request for external_data rego function.
+type RegoRequest struct {
+	ProviderName string        `json:"provider"`
+	Keys         []interface{} `json:"keys"`
+}
+
+// ProviderRequest is the API request for the external data provider.
+type ProviderRequest struct {
+	APIVersion string  `json:"apiVersion,omitempty"`
+	Kind       string  `json:"kind,omitempty"`
+	Request    Request `json:"request,omitempty"`
+}
+
+// Request is the struct that contains the keys to query.
+type Request struct {
+	Keys []interface{} `json:"keys,omitempty"`
+}
+
+// NewRequest creates a new request for the external data provider.
+func NewProviderRequest(keys []interface{}) *ProviderRequest {
+	return &ProviderRequest{
+		APIVersion: "externaldata.gatekeeper.sh/v1",
```

Comment:
Do we want to start with `v1` here since this is the contract between Gatekeeper and providers? v1 signals stable?

Reply:
good catch, updated

Reply:
External data is an alpha feature, right? How do we know the API is stable?

Reply:
But we're already returning `v1` here... are we saying the wire format is stable?

Reply:
That was a mistake, request is updated to be `v1alpha1`

Reply:
My comment above is saying we should not start with `v1`.

Reply:
Ah, oops, I read the comment the wrong way, sorry!